curvefs/test/mds/metaserver_balance_poc.cpp

```diff
@@ -0,0 +1,496 @@
+                dynamic_cast<ChangePeer *>(op.step.get());
+            if (changePeerOp != nullptr) {
+                std::set<MetaServerIdType> members =
+                    topoCopyset.GetCopySetMembers();
+                members.erase(changePeerOp->GetOldPeer());
+                members.insert(changePeerOp->GetTargetPeer());
+                topoCopyset.SetCopySetMembers(members);
+                topology_->UpdateCopySetTopo(topoCopyset);
+
+                // update target usage
+                MetaServer ms;
+                topology_->GetMetaServer(changePeerOp->GetTargetPeer(), &ms);
+                MetaServerSpace space = ms.GetMetaServerSpace();
+                space.SetDiskUsed(space.GetDiskUsed() + 1);
+                topology_->UpdateMetaServerSpace(space,
+                                                 changePeerOp->GetTargetPeer());
+
+                // update source usage
+                topology_->GetMetaServer(changePeerOp->GetOldPeer(), &ms);
+                space = ms.GetMetaServerSpace();
+                space.SetDiskUsed(space.GetDiskUsed() - 1);
+                topology_->UpdateMetaServerSpace(space,
+                                                 changePeerOp->GetOldPeer());
+            }
+
+            schedule::CopySetInfo copyset;
+            ASSERT_TRUE(
+                topoAdapter_->CopySetFromTopoToSchedule(topoCopyset, &copyset));
+            CopySetConf conf;
+            ASSERT_FALSE(opController_->ApplyOperator(copyset, &conf));
+        }
+    }
+
+    void Schedule() {
+        while (true) {
+            int ret = recoverScheduler_->Schedule();
+            std::vector<Operator> ops = opController_->GetOperators();
+            ASSERT_EQ(ret, ops.size());
+            Apply(ops);
+            ASSERT_EQ(0, opController_->GetOperators().size());
+
+            if (ret == 0) {
+                break;
+            }
+        }
+        while (true) {
+            int ret = copysetScheduler_->Schedule();
+            std::vector<Operator> ops = opController_->GetOperators();
+            ASSERT_EQ(ret, ops.size());
+            Apply(ops);
+            ASSERT_EQ(0, opController_->GetOperators().size());
+
+            if (ret == 0) {
+                break;
+            }
+        }
+    }
+
+ protected:
+    std::shared_ptr<DefaultIdGenerator> idGenerator_;
+    std::shared_ptr<MockTokenGenerator> tokenGenerator_;
+    std::shared_ptr<MockStorage> storage_;
+    std::shared_ptr<TopologyImpl> topology_;
+    std::shared_ptr<MockMetaserverClient> mockMetaserverClient_;
+    std::shared_ptr<TopologyManager> serviceManager_;
+
+    std::shared_ptr<ScheduleMetrics> metric_;
+    std::shared_ptr<TopoAdapterImpl> topoAdapter_;
+    std::shared_ptr<OperatorController> opController_;
+    std::shared_ptr<RecoverScheduler> recoverScheduler_;
+    std::shared_ptr<CopySetScheduler> copysetScheduler_;
+};
+
+TEST_F(MetaserverBalancePOC, test_CreatePartitionWithAvailableCopyset_Success) {
```

Comment:
Describe the topo structure of this use case and the topo after the algorithm works.

Reply:
done